src/CommunityToolkit/Xamarin.CommunityToolkit/Views/BadgeView/BadgeView.shared.cs

```diff
@@ -67,7 +67,11 @@ public bool AutoHide
 			set => SetValue(AutoHideProperty, value);
 		}
 
-		static async void OnAutoHideChanged(BindableObject bindable, object oldValue, object newValue) => await (bindable as BadgeView)?.UpdateVisibilityAsync();
+		static async void OnAutoHideChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			if (bindable is BadgeView badgeView)
```

Comment:
I think that we should do a cast, to throw an InvalidCastException, this is the Forms pattern.